text/0006-sampling.md

```diff
@@ -0,0 +1,156 @@
+ future flags, etc.:
+   * Trust parent sampling decision (trusting & propagating parent SpanContext SampleBit)
+   * Always keep
+   * Never keep
+   * Keep with 1/N probability
+
+### Application developer
+These are the folks we've been thinking the most about for OTel in general.
+
+ * They have full control over the OTel implementation or SDK configuration. When using the SDK they
+ can configure custom exporters, custom code/samplers, etc.
+ * They can choose to implement runtime configuration via a variety of means (e.g. baking in feature
+ flags, reading YAML files, etc.), or even configure the library in code.
+ * They make heavy usage of OTel for instrumenting application-specific behavior, beyond what may be
+ provided by the libraries they use such as gRPC, Django, etc.
+
+**Solution:**
+ * Allow application developers to link in custom samplers or write their own when using the
+ official SDK.
+   * These might include dynamic per-field sampling to achieve a target rate
+   (e.g. https://github.com/honeycombio/dynsampler-go)
+ * Sampling decisions are made within the start Span operation, after attributes relevant to the
+ span have been added to the Span start operation but before a concrete Span object exists (so that
+ either a NoOpSpan can be made, or an actual Span instance can be produced depending upon the
+ sampler's decision).
+ * Span.IsRecording() needs to be present to allow costly span attribute/log computation to be
+ skipped if the span is a NoOp span.
+ 
+### Application operator
+Often the same people as the application developers, but not necessarily
+ 
+ * They care about adjusting sampling rates and strategies to meet operational needs, debugging,
+ and cost.
+ 
+**Solution:**
+ * Use config files or feature flags written by the application developers to control the
+ application sampling logic.
+ * Use the config files to configure libraries and infrastructure package behavior.
+
+### Telemetry infrastructure owner
+They are the people who provide an implementation for the OTel API by using the SDK with custom
+`Exporter`s, `Sampler`s, hooks, etc. or by writing a custom implementation, as well as running the
+infrastructure for collecting exported traces.
+
+ * They care about a variety of things, including efficiency, cost effectiveness, and being able to
+ gather spans in a way that makes sense for them.
+
+**Solution:**
+ * Infrastructure owners receive information attached to the span, after sampling hooks have already
+ been run.
+
+## Internal details
+The interface for the Sampler class takes in:
+ * `TraceID`
+ * `SpanID`
+ * Parent `SpanContext` if any
+ * `Links`
+ * Initial set of `Attributes` for the `Span` being constructed
+
+It produces as an output:
+* A boolean indicating whether to sample or drop the span.
+* The new set of initial span Attributes (or passes along the SpanAttributes unmodified)
+* (under discussion in separate RFC) the SamplingRate float.
+
+## Trade-offs
```

Comment:
These read more like "alternatives considered", rather than "trade-offs". Either way, would be useful to capture the reasoning  and pros/cons of these alternatives.

Reply:
I think I re-wrote this section, please take another look. I will close this conversation since it is not relevant anymore but you can start new conversations.

Thanks for your comment.